### DISCUSSION TITLE
force fresh install (clear cache) of single package from tarball

### DISCUSSION THREAD

**Discussion by theoephraim** [pnpm/pnpm#8568] [Q&A]
I am using [pkg.pg.new](https://pkg.pr.new/) to test out preview releases of packages. This means that I'm installing packages from a remote URL related to the PR (ex: `https://pkg.pr.new/@myorg/sometool@123`) which points to a tarball. As my PR is updated, the remote tarball will be updated, but it seems that pnpm wants to cache it locally and will not actually re-install the updated tarball.

Maybe I'm missing something in the docs, but is there a simple way to force a fresh install of a specific package?

Things that do not work:
- `pnpm i`
- `pnpm i --force`
- `pnpm i https://pkg.pr.new/@myorg/sometool@162 --force`
- `pnpm cache delete @myorg/sometool && pnpm i`
- `pnpm cache delete https://pkg.pr.new/@myorg/sometool@162 && pnpm i`

The only thing that does work is clearing the entire cache -- `pnpm store prune && rm -rf node_modules && pnpm i`
But this is obviously undesirable, as it clears the _entire_ pnpm store. I just want to re-install this specific package.

**Comment by theoephraim** [comment]
I see that `pnpm cache delete` was [added recently](https://github.com/pnpm/pnpm/pull/8512), but either it does not fix the issue, or I am using it incorrectly. Perhaps @zkochan can help shed some light?

**Reply by nanarino** [reply]
```json
{
  "dependencies": {
    "@nanarinostyl/lit": "github:nanarino/na-lit",
  }
}
```
i use `pnpm update @nanarinostyl/lit`  it works successfully

**Comment by theoephraim** [comment]
Still seems to be an issue. Anyone have an idea of what is going on?

The only workaround for now is to use the commit hash rather than the PR number, but this is quite cumbersome in an already very awkward workflow...
